src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -37,13 +38,27 @@
    * @param symbols the list of symbols to get a last traded price for.
    * @return a list of last traded price objects for each Symbol that is passed in.
    */
-  public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final List<String> symbols) {
+  public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final String token,
+      final List<String> symbols
+      ) {
     if (CollectionUtils.isEmpty(symbols)) {
       return Collections.emptyList();
     } else {
-      return iexClient.getLastTradedPriceForSymbols(symbols.toArray(new String[0]));
+      return iexClient.getLastTradedPriceForSymbols(token, symbols.toArray(new String[0]));
     }
   }
 
+  /**
+   * Get historical prices given stock and time range.
+   *
+   * @param symbol stock symbol for historical prices
+   * @param range time range for historical prices (e.g. 3m, 6m, 5y)
+   * @return a list of historical prices for the symbol/range passed in
+   */
+  public List<IexHistoricalPrice> getHistoricalPrices(final String token, final String symbol,
+      final String range, final String date) {
+    if (date == null) { return iexClient.getHistoricalPrices(token, symbol, range); }
```

Comment:
I added and changed the logic in IexService to handle optional range and date parameters. Since the IEX API handles both null and empty strings for range/date parameters identically, I changed this method to pass an empty string to IexClient in both cases. I also added some test cases to demonstrate expected results from optional range and date parameters. 

As for symbol, I think my endpoint is handling null or empty symbol strings appropriately. I added test cases to demonstrate expected results from null and empty symbol strings. 

Given the above changes, tests still pass:

![tests_passed](https://user-images.githubusercontent.com/92825615/157898128-39017f52-d226-4cdc-a42c-4ab86e089844.JPG)